src/components/TodoHeader.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from 'react';
+import cn from 'classnames';
+import { Error, Todo } from '../types/Todo';
+import * as todosService from '../api/todos';
+
+type Props = {
+  todos: Todo[],
+  setTodos: (value: Todo[]) => void,
+  tempTodo: Todo | null,
+  setTempTodo: (value: null | Todo) => void,
+  setHasError: (value: Error) => void,
+  loadingIds: number[],
+  setLoadingIds: React.Dispatch<React.SetStateAction<number[]>>,
+};
+
+export const TodoHeader: React.FC<Props> = ({
+  todos,
+  setTodos,
+  tempTodo,
+  setTempTodo,
+  setHasError,
+  loadingIds,
+  setLoadingIds,
+}) => {
+  const [title, setTitle] = useState('');
+
+  const USER_ID = 11041;
+  const neededStatus = !todos.every(todo => todo.completed);
+
+  const toggleAllTodos = async () => {
+    const arrayTodoToUpdate = todos.filter(todo => {
+      if (todo.completed === neededStatus) {
+        return false;
+      }
+
+      return true;
+    });
+
+    setLoadingIds(arrayTodoToUpdate.map(i => i.id));
+
+    await Promise.all(arrayTodoToUpdate.map(async (todo) => {
+      try {
+        await todosService.updateTodo(todo.id, { completed: neededStatus });
+      } catch (err) {
+        setHasError(Error.UPDATE);
+      }
+    }));
+
+    setTodos(todos.map(t => ({
+      ...t,
+      completed: neededStatus,
```

Comment:
Noticed an issue that the todos list is being modified both in case of an error and without it, but it should only change if the operation is successful. I think this case should be moved inside the try block.